python/sglang/lang/interpreter.py

```diff
@@ -588,7 +588,8 @@ def _execute_gen(self, expr: SglGen):
                 else:  # Speculative execution on models with completion interface
                     comp, meta_info = self._spec_gen(sampling_params)
 
-            self.text_ += comp
+            if isinstance(comp, list):
+                self.text_ += comp[0]
```

Comment:
Do we need an `else` for this?

Reply:
To make this PR small, I want to keep `self.text_` just as the first completion from API endpoints. `self.text_` is used in other functions of `interpreter.py` such as fork or speculative generation or select. To keep things simple, these functions won't consider other completions. Instead, it would still stick to the first completion.

Reply:
Is `comp` always an `list` instance? If so, maybe we can raise an error in the `else` condition.